pyquil/api/_compiler.py

```diff
@@ -136,32 +136,31 @@ def to_expression(rule: str) -> ExpressionDesignator:
             _memory=nq_program._memory.copy(),
         )
 
-    def _get_calibration_program(self) -> Program:
+    def _fetch_calibration_program(self) -> Program:
         with self._qcs_client() as qcs_client:  # type: httpx.Client
             response = get_quilt_calibrations(client=qcs_client, quantum_processor_id=self.quantum_processor_id).parsed
         return parse_program(response.quilt)
 
-    def refresh_calibration_program(self) -> None:
-        """Refresh the calibration program cache."""
-        self._calibration_program = self._get_calibration_program()
-
-    @property
-    def calibration_program(self) -> Program:
+    def get_calibration_program(self, refresh: bool = False) -> Program:
         """
         Get the Quil-T calibration program associated with the underlying QPU.
 
+        This will retrieve the current calibration program for the QPU from the
```

Comment:
IMO this first sentence (specifically "on demand") is misleading without a prior mention of the use of a cache.

> Return a cached copy of the calibration program if present; otherwise fetch and store the program from the QCS API.
>
> :param force_refresh: If True, fetch and store a new cached copy of the program even if one is already cached.

^ I'd also favor `force_refresh` which seems more semantically accurate than `refresh`. I.e., even if `refresh=False`, it may still "refresh" if no cached data is present. However, `force_refresh=False` still semantically allows a refresh to happen if needed.

Reply:
Good points.

Reply:
Love the simpler wording